lib/stores/S3Store.js

```diff
@@ -505,14 +505,30 @@ class S3Store extends DataStore {
             this._getMetadata(file_id)
                 .then((metadata) => {
                     return this._retrieveParts(file_id)
-                        .then((parts) => {
-                            return {
-                                parts,
-                                metadata,
-                            };
-                        })
+                        .then((parts) => ({ parts, metadata }))
                         .catch((err) => {
-                            throw err;
+                            // getOffset is called from both the HEAD request and the PATCH request
+                            // (via the _processUpload method). The HEAD request calls it (implicitly)
+                            // with with_parts: false, the _processUpload method calls it (explicitly)
+                            // with with_parts: true. The _processUpload method expects upload parts to be returned,
+                            // which would not work when the upload is completed, since there are no parts anymore.
+                            // TODO: refactor to separate these concerns.
+                            // TODO: write a test for this case.
+                            if (with_parts === true || err !== 'NoSuchUpload') {
```

Comment:
One thing I'm not sure about is whether we can compare `err` with a string. If it an `Error` object, this won't work. (I'm rewriting this to TS in the nearby future with should help with these silly things).

Reply:
I don't think this will work, since the error returned is an object that contains a property `code` with the value 'NoSuchUpload`